Issue702 single zone commercial adrenalin update

dhblum opened the pull request:
This is for #702, following from PR #703 to finalize.  This is also for #635, #432, and #733.

dhblum commented:
@HWalnum This PR is almost finished.  I didn't see the updates you made to the`` releasenotes.md`` previously but I added them and also marked  the changes in the KPI results for all the scenarios, which I think is good to track if they change significantly version to version.  Can you look at the KPI changes I marked and double check they are what you'd expect or experienced based on your changes?  https://github.com/ibpsa/project1-boptest/pull/734/files#diff-bf0fb780cd5da59cf3fcbb2aeac981b23584b3089c661658f30b386ff4440229.

HWalnum commented:
Hi @dhblum. For the peak_heat scenario looks reasonable. The changes in the typical_heat scenario seems a bit drastic. I think I might have not updated the days after switching back to the original weather data. However, I tried to have a look with the find_days algorithm. Looking at the result I got from that, I wonder if the median of the daily max is reasonable for commercial buildings with much lower consumption in weekends. This makes the whole algorithm a bit random. 

In the Adrenalin competition we selected the days based more on visual inspection, because the algorithm ending putting one of the scenarios during a holiday, which gave different occupancy profiles.

dhblum commented:
@HWalnum Hm ok on the day selection algorithm, thanks for reporting.  It certainly isn't as bullet-proof as we thought in the beginning, various issues have come up over time.  Perhaps there can be another option in the script to id and remove weekends.  At least it's documental in terms of the options chosen to produce the days.  Or, we just say that the given days are the time period for a given scenario because the test case developer thought it was reasonable for intent of the scenario.

HWalnum commented:
I definetly see the benefit of having a documentable procedure, but it is challenging to make it fit all cases. For typical heat day, it could be an idea to look at weekly averages. However, it will not remove the risk of including special holdiays etc. However, the last is mainly a problem for cases like this, where real measured occupancy is used. 

For this particular case, I would suggest to go for 308 as typical_heat_day. This is based on median of weekly averages, and it should not coincide with any holidays.

dhblum commented:
Ok thanks @HWalnum.  Then 308 sounds good and I will plan to prep the PR using that.

dhblum commented:
Hey @HWalnum I've updated the days.json with the new typical heat period, updated unit test ref results, and updated the KPI changes.  Take a look at https://github.com/ibpsa/project1-boptest/pull/734/commits/456d885b906c17e64f55234f8244f55b9077e3f3 and see if the new change values are reasonable to you.  I think they seem ok.  Otherwise, just waiting for unit tests to pass.

HWalnum commented:
@dhblum Looks good to me :)

dhblum commented:
Great thanks.  Once unit tests pass, I'll push the new test case FMU and finalize for merging.

Thank you so much for your help and effort on this.

HWalnum commented:
Shit. Sorry about that. 
The changes should have been done in DistrictHeating2. It was only adding the two read blocks (+ modifying the description in the overwrite block). So it should not require changes to the documentation. 
I should have deleted the old models from the repository. Maybe we should do that now.
Do you want me to do the changes, or do you want to do it yourself, in you PR?

dhblum commented:
No worries.  Can you make a PR to branch ``issue702_single_zone_commercial_adrenalin_update`` with the changes to the Modelica models, including deleting old models to prevent confusion in the future?  

I already committed changes to the docs that used the (now incorrect) recompiled FMU (I noticed something that made me want to do that).  It's ok, after your new PR, I'll re-update things, and thereby double checking it's all consistent.

dhblum commented:
Alright @HWalnum thanks.  I merged PR https://github.com/ibpsa/project1-boptest/pull/737 and updated docs, bacnet.ttl, and unit test results.  Just waitting for unit tests to pass and then will again update wrapped FMU and mo.  Then we should be ready to merge :)

Thanks again.
